web/src/main/java/org/springframework/security/web/password/WellKnownChangePasswordRedirectFilter.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2002-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.web.password;
+
+import org.springframework.security.web.DefaultRedirectStrategy;
+import org.springframework.security.web.RedirectStrategy;
+import org.springframework.security.web.util.matcher.AntPathRequestMatcher;
+import org.springframework.security.web.util.matcher.RequestMatcher;
+import org.springframework.util.Assert;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+/**
+ * Redirect filter for a Well-Known URL for Changing Passwords.
+ *
+ * @author Evgeniy Cheban
+ * @since 5.4
+ */
+public class WellKnownChangePasswordRedirectFilter extends OncePerRequestFilter {
```

Comment:
Since there's nothing really specific to changing passwords in this filter, let's instead call it `RequestMatcherRedirectFilter`. We'd also want to move it to `web` in that case.

I imagine the constructor would change to take both a `RequestMatcher` and a redirect location.

Since this would now be a more generic filter, I think we should also remove it from the `FilterComparator` and instead have the DSL use `addFilterBefore`.

Reply:
Done.

Reply:
@jzheaux I did the same refactoring for the `WebFilter`.